SD_Card_Driver/Core/Inc/LOS_D_sd.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef LOS_D_SD_H_
+#define LOS_D_SD_H_
+
+#include "fatfs.h"
+#include "string.h"
+#include "LOS_D_storage_device.hpp"
+
+class SDCard : StorageDevice {
+	private:
+		FATFS fs;
+		const char* filePath;
+		bool fileOpened = false;
+		FIL file;
+
+		int open(const char* location);
+
+	public:
+		int init() override;
+		bool checkExist(const char* location);
+		int read(const char* location, uint8_t* data, size_t size, size_t offset = 0) override;
+		int write(const char* location, uint8_t* data, size_t size, size_t offset = 0) override;
+		size_t length(const char* location) override;
+		int cleanup() override;

```

Comment:
so I guess this goes for the storage device interface too... for the return types, do we need to use `int` return types? In embedded we try to be more specific with the size of the data type bec depending on the architecture, an int could be different sizes and also we may not need the entire int storage (4 bytes most of the time) would an `int8_t` or `uint8_t` data type do? In this case it is explicit how large our value is and also if it could be signed or unsigned.

Reply:
I assume what is being returned is a some error code so I think either one of the options above should do

Reply:
Yes, the error code from FatFS is what is being returned. `uint8_t` should be big enough to cover all the error codes since it only goes up to 19.

Reply:
it should be unsigned unless there's a need for a signed integer.